runtime/queries/zig/highlights.scm

```diff
@@ -182,123 +210,105 @@
-
-; Modules
-
-(variable_declaration
-  (identifier) @variable ; TODO: module
-  (builtin_function
-    (builtin_identifier) @keyword.control.import
-    (#any-of? @keyword.control.import "@import" "@cImport")))
-
-(variable_declaration
-  (identifier) @variable ; TODO: module
-  (field_expression
-    object: (builtin_function
-      (builtin_identifier) @keyword.control.import
-      (#any-of? @keyword.control.import "@import" "@cImport"))))
-
-; Functions
-
-(call_expression
-  function: (field_expression
-    member: (identifier) @function.method))
-
-(call_expression
-  function: (identifier) @function)
-
-(function_declaration
-  name: (identifier) @function)
-
-(builtin_identifier) @function.builtin
-
-; Fields
-
-(field_initializer
-  .
-  (identifier) @variable.other.member)
-
-(field_expression
-  (_)
-  member: (identifier) @type (#match? @type "^[A-Z_][a-zA-Z0-9_]*"))
-
-(field_expression
-  (_)
-  member: (identifier) @variable.other.member)
-
-(container_field
-  name: (identifier) @variable.other.member)
-
-(initializer_list
-  (assignment_expression
-      left: (field_expression
-              .
-              member: (identifier) @variable.other.member)))
-
-; Labels
-
-(block_label (identifier) @label)
-
-(break_label (identifier) @label)
+  "="
+  "*="
+  "*%="
+  "*|="
+  "/="
+  "%="
+  "+="
+  "+%="
+  "+|="
+  "-="
+  "-%="
+  "-|="
+  "<<="
+  "<<|="
+  ">>="
+  "&="
+  "^="
+  "|="
+  "!"
+  "~"
+  "-"
+  "-%"
+  "&"
+  "=="
+  "!="
+  ">"
+  ">="
+  "<="
+  "<"
+  "&"
+  "^"
+  "|"
+  "<<"
+  ">>"
+  "<<|"
+  "+"
+  "++"
+  "+%"
+  "-%"
+  "+|"
+  "-|"
+  "*"
+  "/"
+  "%"
+  "**"
+  "*%"
+  "*|"
+  "||"
+  ".*"
+  ".?"
+  "?"
+  ".."
+] @operator
 
-; Constants
+; Literals
 
-((identifier) @constant
-  (#match? @constant "^[A-Z][A-Z_0-9]+$"))
+(character) @constant.character
 
 [
-  "null"
-  "undefined"
-] @constant.builtin
+  (string)
+  (multiline_string)
+] @string
 
-(field_expression
-  .
-  member: (identifier) @constant)
+(integer) @constant.numeric.integer
 
-(enum_declaration
-  (container_field
-    type: (identifier) @constant))
+(float) @constant.numeric.float
 
-; Types
+(boolean) @constant.builtin.boolean
 
-(parameter
-  type: (identifier) @type)
+(escape_sequence) @constant.character.escape
 
-((identifier) @type
-  (#match? @type "^[A-Z_][a-zA-Z0-9_]*"))
+; Punctuation
 
-(variable_declaration
-  (identifier) @type
-  "="
-  [
-    (struct_declaration)
-    (enum_declaration)
-    (union_declaration)
-    (opaque_declaration)
-  ])
+[
+  "["
+  "]"
+  "("
+  ")"
+  "{"
+  "}"
+] @punctuation.bracket
 
 [
-  (builtin_type)
-  "anyframe"
-] @type.builtin
+  ";"
+  "."
+  ","
+  ":"
+  "=>"
+  "->"
+] @punctuation.delimiter
 
-; Parameters
+(payload "|" @punctuation.bracket)
 
-(parameter
-  name: (identifier) @variable.parameter)
+; Comments
 
-; Variables
+((comment) @comment.block.documentation
+  (#match? @comment.block.documentation "^//!"))
 
-(identifier) @variable
+(comment) @comment.line
```

Comment:
I think these two should be reversed? Now the less specific highlight should be higher in the file - I think `comment.line` will always overwrite `comment.block.documentation` here

Reply:
ah yeah missed that fixed